Sources/SwiftInspectorVisitors/StructVisitor.swift

```diff
@@ -101,9 +106,18 @@ public final class StructVisitor: SyntaxVisitor {
   }
 
   public override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
-    if !hasFinishedParsingStruct, let _ = structInfo {
+    if !hasFinishedParsingStruct, let structInfo = structInfo {
       // We've previously found a struct declaration, so this must be an inner enum.
-      // TODO: Utilize enum visitor to find inner enum information, utilizing parent information from structInfo
+      let qualifiedParentTypeName = QualifiedParentNameCreator.createNameGiven(
+        currentParentTypeName: parentTypeName,
+        currentTypeName: structInfo.name)
+
+      let enumVisitor = EnumVisitor(parentTypeName: qualifiedParentTypeName)
+      enumVisitor.walk(node)
+      innerEnums += enumVisitor.enums
+      innerClasses += enumVisitor.innerClasses
+      innerStructs += enumVisitor.innerStructs
```

Comment:
It feels like it would be really easy to miss one line here... I wonder if we want a way to generate all valid permutations in our tests to ensure that we cover all cases now and in the future?

Reply:
Or maybe it's easy enough to write them all out, as it seems like we are doing now.

Reply:
I do want to consolidate at some point in the future. At a minimum we have good tests so we should be catch when we screw this up. I've been focused on getting something working this last week rather than generalizing the reused code – there's certainly room to go back and make this code less repetitive. Not sure how to do that just yet though.

Reply:
Seems like we are doing the right thing for now.